nifi-extension-bundles/nifi-standard-services/nifi-lookup-services-bundle/nifi-lookup-services/src/test/java/org/apache/nifi/lookup/maxmind/TestIPLookupService.java

```diff
@@ -0,0 +1,198 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.nifi.lookup.maxmind;
+
+import com.maxmind.geoip2.DatabaseReader;
+import org.apache.nifi.lookup.TestProcessor;
+import org.apache.nifi.serialization.record.Record;
+import org.apache.nifi.util.TestRunner;
+import org.apache.nifi.util.TestRunners;
+import org.jetbrains.annotations.NotNull;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.InetAddress;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Collections;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+public class TestIPLookupService {
+    private static final String SOME_VALID_IP = "0.0.0.0";
+
+    @TempDir
+    static Path tempDir;
+    static Path tempDummyMmdbFile;
+
+    private TestRunner runner;
+    private IPLookupService testSubject;
+
+    private DatabaseReader mockDatabaseReader;
+
+    @BeforeAll
+    public static void init() throws IOException {
+        tempDummyMmdbFile = Files.createFile(tempDir.resolve("dummy.mmdb"));
+    }
+
+    @BeforeEach
+    public void setUp() {
+        runner = TestRunners.newTestRunner(TestProcessor.class);
+        testSubject = new IPLookupService() {
+            @NotNull
+            @Override
+            DatabaseReader createDatabaseReader(File dbFile) throws IOException {
+                return mockDatabaseReader;
+            }
+        };
+
+        mockDatabaseReader = mock(DatabaseReader.class);
+    }
+
+    @Test
+    void testLookupDefaultNoResult() throws Exception {
+        // GIVEN
+        runner.addControllerService("testSubject", testSubject);
```

Comment:
Recommend creating a static value for `testSubject` and reusing across the test methods.